packages/core/core/src/services/document-service/internationalization.ts

```diff
@@ -69,7 +54,7 @@ const localeToData: Transform = (contentType, params) => {
     return params;
   }
 
-  if (params.locale) {
+  if (params.locale && typeof params.locale === 'string' && params.locale !== '*') {
     return assoc(['data', 'locale'], params.locale, params);
   }
```

Comment:
should we throw an error if users try to use a locale that's invalid?

What happens if a user is creating a document but sends a locale that's not a string?